serving/docs/lmi/tutorials/trtllm_manual_convert_tutorial.md

```diff
@@ -0,0 +1,264 @@
+
+## Step by step tutorial
+
+In this tutorial, we will be converting the baichuan model to TensorRT-LLM model format on p4d.24xlarge.
+
+### Step 1: Choose your instance
+
+To do the model compilation you need to use the instance that has **the same GPU architecture** that will be used for your inference deployment (e.g model compiled on a g5 instance can be deployed only on g5 instances).
+
+### Step 2: Pull the docker image
+
+Refer [here](https://github.com/aws/deep-learning-containers/blob/master/available_images.md#large-model-inference-containers) for the latest TensorRT-LLM DLC and pull the image.
+
+For example:
+
+```
+docker pull 763104351884.dkr.ecr.us-west-2.amazonaws.com/djl-inference:0.25.0-tensorrtllm0.5.0-cu122
+```
+
+You can also pull the container from DockerHub:
+
+```
+docker pull deepjavalibrary/djl-serving:0.25.0-tensorrt-llm
+```
+
+### Step 3: Login the container and prepare the environment
+
+You need to manually login into the container to proceed for conversion
+
+```
+docker run -it --runtime=nvidia --gpus all \
+--shm-size 12g \
+deepjavalibrary/djl-serving:0.25.0-tensorrt-llm \
+/bin/bash
+```
+
+Then clone the TensorRT-LLM repository inside the container for manual conversion.
+Please make sure the version of the TensorRT-LLM need to match the one installed on the system.
+
+You can check it with
+
+```
+pip show tensorrt-llm
+# Output
+# Name: tensorrt-llm
+# Version: 0.5.0
+```
+
+Then just clone the TensorRT-LLM Triton backend for model preparation. If the version is 0.5.0, then you need to checkout the tag for that version (v0.5.0).
+```
+git clone https://github.com/triton-inference-server/tensorrtllm_backend -b v0.5.0
+cd tensorrtllm_backend && rm -rf tensorrt_llm
+git clone https://github.com/NVIDIA/TensorRT-LLM -b v0.5.0 tensorrt_llm
+```
+
+
+### Step 4: Build TensorRT-LLM compatible model
+
+The following work we do is very similar to Triton Server model preparation. You can find more information on the [official preparation doc](https://github.com/triton-inference-server/tensorrtllm_backend/tree/release/0.5.0?tab=readme-ov-file#prepare-tensorrt-llm-engines).
+Our environment is pre-configured all necessary packages, so we don't need to pip install anything.
+
+Here we just need to
+
+```
+cd tensorrt_llm/examples/baichuan
+# or just tensorrt_llm/examples/<model you like>
+```
+
+Checking the `README.md` we will find the following instruction. We just need to add a few more parameter to build ready-to-go model:
+
+```
+# Build the Baichuan V1 13B model using a single GPU and FP16.
+python build.py --model_version v1_13b \
+                --model_dir baichuan-inc/Baichuan-13B-Chat \
+                --dtype float16 \
+                --use_gemm_plugin float16 \
+                --use_gpt_attention_plugin float16 \
+                --world_size 2 \
+                --output_dir baichuan_v1_13b/trt_engines/fp16/2-gpu/ \
+                --use_inflight_batching
```

Comment:
let's also add `--enable_context_fmha`, `--remove_input_padding` and `--parallel_build` as those args are always recommended.